build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -91,7 +91,7 @@ private void generateJavadocs(Project project, ProjectLayout layout, TaskContain
             // TODO: This breaks the provider
             options.links(javadocs.getJavaApi().get().toString(), javadocs.getGroovyApi().get().toString());
 
-            task.source(extension.getDocumentedSource());
+            task.source(extension.getDocumentedSource().filter(f -> f.getName().endsWith(".java")));
```

Comment:
I'm not sure about this.  Did javadoc not fail when it groovy and kt files were present? Why does it matter that resources is on it now?

This seems very smelly since it's a filter being put on top of what should be the "correct" set of files.